card/db/migrate_core_cards/20150202143810_import_bootstrap_layout.rb

```diff
@@ -11,28 +11,42 @@ def up
         
     import_json "bootstrap_layout.json"#, :pristine=>true, :output_file=>nil
 
-#    if unmerged.empty?
-      if layout && layout.pristine? &&
-        all = Card[:all]
-        layout_rule_card = all.fetch :trait=>:layout
-        style_rule_card  = all.fetch :trait=>:style
-        if layout_rule_card.pristine? && style_rule_card.pristine?
-          layout_rule_card.update_attributes! :content=> '[[Default Layout]]'
-          style_rule_card. update_attributes! :content=> '[[classic bootstrap skin]]'
+    if layout && layout.pristine? &&
+      all = Card[:all]
+      layout_rule_card = all.fetch :trait=>:layout
+      style_rule_card  = all.fetch :trait=>:style
+      if layout_rule_card.pristine? && style_rule_card.pristine?
+        layout_rule_card.update_attributes! :content=> '[[Default Layout]]'
+        if style_rule_card.item_names.first == 'customized classic skin'
+          Card.create! :name=>'customized bootstrap skin', :type=>'Skin', 
+            :content=>"[[classic bootstrap skin]]\n[[*css]]"
+          style_rule_card.update_attributes! :content=> '[[customized bootstrap skin]]'
+        else
+          style_rule_card.update_attributes! :content=> '[[classic bootstrap skin]]'
         end
       end
-#    else
-#      unmerged.map! do |row|
-#        puts "didn't merge #{row['name']}"
-#        row['name'] += '+alternate'
-#        row
-#      end
-#      Card.merge_list unmerged
-#    end
+    end
+    
+    Card.create! :name=>"*header+*self+*read", :content=>'[[Anyone]]'
+    
+    # merge "style: functional" and "style: standard" into "style: cards"
+    old_func = Card[:style_functional]
+    old_func.name = 'style: cards'
+    old_func.codename = 'style_cards'
+    old_func.update_referencers = true
+    old_func.save!
+    
+    old_stand = Card[:style_standard]
+    old_stand.codename = nil
+    old_stand.delete!
+    
 
     # these are hard-coded
-    Card.create! :name=>'style: bootstrap theme', :type_code=>:css, :codename=>'bootstrap_theme_css'
-    Card.create! :name=>'style: bootstrap', :type_code=>:css, :codename=>'bootstrap_css'
+    Card.create! :name=>'theme: bootstrap_default', :type_code=>:css, :codename=>'theme_bootstrap_default'
+    Card.create! :name=>'style: bootstrap',         :type_code=>:css, :codename=>'bootstrap_css'
+    Card.create! :name=>'style: bootstrap cards',   :type_code=>:css, :codename=>'bootstrap_cards'
+    
+    Card.create! :name=>'style: bootstrap compatible', :type_code=>:scss, :codename=>'style_bootstrap_compatible'
     Card.create! :name=>'script: bootstrap', :type_code=>:js, :codename=>'bootstrap_js'
```

Comment:
handling pretty much everything in bootstrap styles in code now.  should make updates easier (and I suspect many are coming)